openapi.yaml

```diff
@@ -164,704 +128,833 @@ paths:
+      suffix:
+        type: string
+        description: Suffix of the supervisor.
+  JobPostRequired:
+    type: object
     required:
       - positionNumber
+      - employmentType
       - beginDate
-      - supervisorOsuID
-      - supervisorPositionNumber
+      - supervisor
       - effectiveDate
+  JobAttributes:
+    type: object
+    additionalProperties: false
     properties:
-      positionNumber:
-        type: string
-        description: Identifies a position
-        example: C12345
       suffix:
         type: string
-        # TODO: find a better way to describe different required fields for
-        # a given set of attributes. Suffix is required for PUT but not POST.
-        description: |
+        description: >
           Used to uniquely identify a job record for persons who have
           multiple jobs that share the same position number.
           For updating existing jobs, this field is required.
+      positionNumber:
+        type: string
+        description: Identifies a position.
+        example: C12345
+      employmentType:
+        type: string
+        enum: *employmentTypeEnums
+        description: Type of employment.
       effectiveDate:
         type: string
         format: date
         description: The date this job assignment is effective.
-        example: '2018-12-31'
+        example: '1970-1-21'
       beginDate:
         type: string
         format: date
         description: Begin date of job.
-        example: '2018-12-31'
+        example: '1970-1-21'
       endDate:
         type: string
         format: date
         description: End date of job.
-        example: '2019-01-01'
-      accruesLeave:
+        example: '1971-2-15'
+      accruesLeaveInd:
         type: boolean
-        description: If true, the job accrues leave.
+        description: If true, the job accrues leave time.
       contractBeginDate:
         type: string
         format: date
         description: The date the contract begins.
-        example: '2018-12-31'
+        example: '1970-1-21'
       contractEndDate:
         type: string
         format: date
         description: The date the contract ends.
-        example: '2018-12-31'
+        example: '1971-2-21'
       contractType:
-        type: string
-        description: Contract type code/ID.
-      locationID:
-        type: string
-        description: Location ID of where job will be performed.
+        type: object
+        properties:
+          code:
+            type: string
+            enum: [P, S, O]
+            description: |
+              Contract type code/Id.
+              - `P` - Primary
+              - `S` - Secondary
+              - `O` - Overload
+          description:
+            type: string
+            description: Description of contract code
+      campus:
+        type: object
+        properties:
+          code:
+            type: string
+            description: Id of campus where this job will be performed.
```

Comment:
For general sentences, I think ID is the correct form. `Id` is only used for the variables naming style.